### PR TITLE
linq_fold: plan_decs_unroll Slice 5f — terminal splice for last/single/aggregate/element_at

### PR DESCRIPTION
**Pull request by borisbat**
## Summary

Closes the 4 outliers identified in PR #2812's m4 bench snapshot. These terminators were falling through to `to_sequence` materialization because `plan_decs_unroll` only recognized `first`/`any`/`all`/`contains`/`min_by`/`max_by` plus the accumulator family. Wave 5 extends the splice path to cover walk-all single-return terminators + `element_at` counter early-exit.

**`emit_decs_walk_lane`** (`last` / `last_or_default` / `single` / `single_or_default` / `aggregate`):
- State (found flag, retained element, accumulator) hoisted at invoke scope so it persists across archetype boundaries.
- `single_or_default` uses `for_each_archetype_find` for early-exit on 2nd match (sets stop flag, returns `true` from inner lambda); others use plain `for_each_archetype` unless `take`/`take_while` forces bool-lambda dispatch.
- `aggregate` mirrors `emit_accumulator_lane`'s workhorse / non-workhorse seed branch (`=` vs `<-`) and peel-or-invoke fallback for the binary lambda.

**`emit_decs_element_at`** (`element_at` / `element_at_or_default`):
- `idx` + counter + `found` + result hoisted at invoke scope.
- `perElement` compares `cnt == idx`, writes result + returns `true` on match (uses `for_each_archetype_find` for the counter early-exit).
- Pre-loop `idx<0` panic (`element_at`) or `return default<T>` (`_or_default`).
- Tail handles "not found" = out-of-range = `idx >= effective length`.

## Bench (INTERP, 100K rows, ns/op)

| Lane | master m4 | branch m4 | speedup | vs m3f |
|---|---|---|---|---|
| `last_match` | 82 | 17 | 4.8× | 17 vs 5 |
| `single_match` | 80 | 14 | 5.7× | 14 vs 3 |
| `aggregate_match` | 82 | 7 | 11.7× | **7 vs 5 ≈ matches** |
| `element_at_match` | 34 | 0 | ∞ | **0 vs 0 matches** |

Per-op allocs drop **84 B → 1 B** across all four — eliminated the `array<tuple>` materialization.

## Test plan

- [x] 16 new tests in `tests/linq/test_linq_from_decs.das` (8 parity + 4 splice-shape gates + 2 range-interaction + 2 multi-match edge cases)
- [x] 185 tests in file (up from 169)
- [x] 1376 `tests/linq` suite — interp + AOT green
- [x] 245 `tests/decs` suite — interp + AOT green
- [x] 371 `tests/ast_match` + 10 `tests/template` + 9 `tests/macro_call` + 27 `tests/macro_boost` green
- [x] Non-outlier bench smoke (`count_aggregate_m4`) unchanged at 6 ns/op
- [x] Lint clean on `daslib/linq_fold.das`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: daslib/linq_fold.das [hunk 1]
```diff
@@ -3900,6 +3900,263 @@ def private emit_decs_min_max_by(bridge : DecsBridgeShape?;
...
+                if (!$i(foundName)) panic("sequence contains no elements")
+                return <- $i(bindName)
+            }
+        } else {
+            let defaultName = qn("decs_dv", at)
+            preludeStmts |> push <| qmacro_expr() {
+                let $i(defaultName) = $e(terminatorCall.arguments[1])
+            }
+            tailStmts |> push_from <| qmacro_block_to_array() {
+                if (!$i(foundName)) return $i(defaultName)
```
[the remark this answers is not in the document]
Comment (borisbat) [reply daslib/linq_fold.das]: Fixed in `9f7786826`. Added two helpers `emit_default_bind` / `emit_default_return` that emit the can_copy / clone_to_move branch matching daslib/linq.das, and applied at all 6 default-return sites (decs `last_or_default` + `single_or_default` + array-side mirror of the same — pre-existing latent bug since no test ever exercised non-copyable defaults). `element_at_or_default` in both paths has no user-supplied default, so it now uses `var t : T; return <- t` (linq.das line 2547's move-init shape) instead of `return default<T>`. Regression covered in new `tests/linq/test_linq_fold_non_copyable_default.das` with `array<int>` defaults. Note: `element_at`'s success path (`return $i(valueName)`) still plain-returns the element — pre-existing on both paths, separate fix needs the same can_copy probe on element-type returns.

File: daslib/linq_fold.das [hunk 2]
```diff
@@ -3900,6 +3900,263 @@ def private emit_decs_min_max_by(bridge : DecsBridgeShape?;
...
+        perElement = qmacro_block() {
+            if ($i(foundName)) {
+                $i(stopName) = true
+                return true
+            }
+            $i(foundName) = true
+            $i(bindName) := $i(finalBind)
+        }
+        tailStmts |> push_from <| qmacro_block_to_array() {
+            if ($i(stopName) || !$i(foundName)) return $i(defaultName)
```
[the remark this answers is not in the document]
Comment (borisbat) [reply daslib/linq_fold.das]: Same fix applied here in `9f7786826` — `single_or_default`'s 2 default-return sites (perElement on 2nd match + tail on empty) now go through `emit_default_return` which routes through `clone_to_move` for non-copyable types. Regression test covers both the empty-source and multi-match paths.
